connectors/agentic-ai/src/main/java/io/camunda/connector/agenticai/model/tool/ToolCall.java

```diff
@@ -10,10 +10,11 @@
 import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
 import io.camunda.connector.agenticai.model.AgenticAiRecord;
 import java.util.Map;
+import javax.annotation.Nullable;
 
 @AgenticAiRecord
 @JsonDeserialize(builder = ToolCall.ToolCallJacksonProxyBuilder.class)
-public record ToolCall(String id, String name, Map<String, Object> arguments)
+public record ToolCall(@Nullable String id, String name, Map<String, Object> arguments)
```

Comment:
💭 I fear Vertex AI not supporting tool call IDs might make some things complicated in the long run. I think it's OK for now, but we might need to think about if it is the right approach to have this nullable (vs. generating an ID on our side if it is missing).

Reply:
I though of generating a random ID but was not sure about the implications.